docs/sql-migration-guide.md

```diff
@@ -24,6 +24,7 @@ license: |
 
 ## Upgrading from Spark SQL 3.5 to 4.0
 
+- Since Spark 4.0, the default behaviour when inserting map elements in `ArrayBasedMapBuilder` is changed to first apply key normalization. To restore the previous behaviour, set `spark.sql.legacy.disableMapKeyNormalization` to `true`.
```

Comment:
let's be more user-facing. The affected SQL functions are `create_map`, `map_from_arrays`, `map_from_entries`, and `map_concat`. `ArrayBasedMapBuilder` is an internal thing and we should not mention it to the end users.

Reply:
Added: 

> The affected SQL functions are create_map, map_from_arrays, map_from_entries, and map_concat.

And removed `ArrayBasedMapBuilder`.